src/enftun.c

```diff
@@ -244,12 +268,16 @@ enftun_connect(struct enftun_context* ctx)
 
     rc = enftun_tunnel(ctx);
 
+ close_conn_state:
```

Comment:
The `conn_state` is prepared before the tun or tls objects, so it should be closed after them.

Reply:
As written, if line 259 is reached, we'll try to close a tun and a tls that were never opened or connected.